### PR TITLE
Propeller speed is computed now

### PR DESCRIPTION
**Pull request by xjjiang**
### Summary

In the current design, propeller speed is a user given input. Now, it is computed from two new variables: `MAX_TIP_SPEED` and `PERCENT_ROTOR_RPM_CORRECTED`.

### Related Issues

- Resolves #

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: aviary/subsystems/propulsion/prop_performance.py
Comment (johnjasa) [inline aviary/subsystems/propulsion/prop_performance.py:201]: Do you expect the this minimum function to switch between the expression or the `max_prop_tip_spd` frequently during an optimization? If so, using a [smooth minimum](https://en.wikipedia.org/wiki/Smooth_maximum) function would improve convergence.

Based on the calculations here, I wouldn't expect using a regular minimum function to be a problem in this case. This is more of a general note and something to consider when using gradient-based optimization.

File: aviary/subsystems/propulsion/test/test_prop_performance.py
Comment (johnjasa) [inline aviary/subsystems/propulsion/test/test_prop_performance.py:104]: Can you remove all of these `PROPELLER_TIP_SPEED` `set_val` calls in this test now?

**Review by johnjasa — APPROVED** [review]
Approving so I don't hold this up! Two quick comments.

File: aviary/subsystems/propulsion/test/test_prop_performance.py
Comment (crecine) [inline aviary/subsystems/propulsion/test/test_prop_performance.py:107]: here and below, should this be `[1,1,1]`?

**Review by crecine — APPROVED** [review]

File: aviary/subsystems/propulsion/prop_performance.py
Comment (xjjiang) [reply aviary/subsystems/propulsion/prop_performance.py:201]: For takeoff, I expect that it is always at maximum tip speed. For cruise, I expect that it is always less than maximum tip speed. I use `maximum()` because it is a built-in function for `ExecComp`. I am not sure if a smooth function can replace it.

File: aviary/subsystems/propulsion/test/test_prop_performance.py
Comment (xjjiang) [reply aviary/subsystems/propulsion/test/test_prop_performance.py:104]: You are right. I removed it.

File: aviary/subsystems/propulsion/test/test_prop_performance.py
Comment (xjjiang) [reply aviary/subsystems/propulsion/test/test_prop_performance.py:107]: I guess we can assume that this parameter varies but in GASP, it is a constant in each phase (takeoff, climb, cruise). I changed it to [1,1,1].

Comment (crecine) [reply aviary/subsystems/propulsion/test/test_prop_performance.py:107]: thanks

**Pull request merged**
